core/modules/rewrite.cc

```diff
@@ -119,10 +120,7 @@ inline void Rewrite::DoRewrite(bess::PacketBatch *batch) {
     bess::utils::CopyInlined(ptr, templates_[start + i], size, true);
   }
 
-  next_turn_ = start + cnt;
-  if (next_turn_ >= bess::PacketBatch::kMaxBurst) {
```

Comment:
I'm fine with this approach, but could the unfairness may actually have come from the fact that we compare `next_turn_` to `PacketBatch::kMaxBurst` here instead of `kNumSlots`, which is set to `bess::PacketBatch::kMaxBurst * 2 - 1`?

Reply:
We can't compare `next_turn_` with `kNumSlots` here; if it is larger, then it implies out-of-bound access happened in the for loop.